### ISSUE TITLE
Input type "email" error

### ISSUE THREAD

**Issue by zslabs** [nosir/cleave.js#313]
https://jsfiddle.net/gLLsrxxf/489/

When using `type="email"` I get the following error: `Failed to execute 'setSelectionRange' on 'HTMLInputElement': The input element's type ('email') does not support selection.` I don't think I'm doing anything out of the ordinary (quickly just added that type to the JS fiddle), but if we're able to check against any/all valid input types, that'd be awesome!

**Comment by zslabs** [comment]
Looking into this a bit further, types text, search, URL, tel and password are the only input types that support `setSelectionRange` https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange

One option could be to check against that and throw up a warning that Cleave will be ignored if attached to a type that doesn't support it.

```
if (elem.setSelectionRange && /text|search|password|tel|url/i.test(elem.type || '')) {
    etc ...
}
```

☝️ from https://github.com/vitalets/x-editable/issues/939

**Comment by nosir** [comment]
Good to know.
So I can add a try catch for `setSelectionRange`, would it solve your problem too?

**Comment by zslabs** [comment]
I think that'd be a great addition. Potentially a `console.warn` may also help devs know "why" something isn't working as expected as well.

**Comment by nosir** [comment]
Released in v1.3.3
